File: layout/macros/examples/label.py
import LayoutScript
from LayoutScript import *

# automatic labeling of an array

   
l=project.newLayout(); 

dr=l.drawing # pointer to the main drawing

dr.currentCell.cellName="python_example_label"

dr.clearPoints()



for x in range (0,10):
        for y in range (0,10):
                dr.point(x*1800,y*800)
                dr.text("#"+str(x)+"/"+str(y))
        


dr.selectAll()
dr.setWidth(400)
dr.toPolygon()
dr.deselectAll();

# writes result to the home folder

import os
l.drawing.saveFile( os.path.expanduser('~')+"/testout.gds")


print ("Python script completed" )

File: layout/macros/examples/label.rb
require 'LayoutScript'
include LayoutScript

# automatic labeling of an array

l=Project.new_layout()

dr=l.drawing # pointer to the main drawing

dr.currentCell.cellName="ruby_example_label"


for x in (0..9)
        for y in (0..9)
                dr.point(x*1800,y*800)
                dr.text("##{x}/#{y}")
        end
end

dr.select_all()
dr.set_width(400)
dr.to_polygon()
dr.deselect_all();


# save to your home folder
l.drawing.save_file("#{Dir.home}/testout.gds")

puts "Ruby script completed" 